tests/test_utils.py

```diff
@@ -60,3 +63,22 @@ def test_to_absolute_path_without_hydra(
     path = str(Path(path))
     expected = str(Path(expected).absolute())
     assert utils.to_absolute_path(path) == expected
+
+
+@mark.parametrize(
+    "env,expected_error",
+    [
+        param({}, False, id="env_unset"),
+        param({"HYDRA_DEPRECATION_WARNINGS_AS_ERRORS": ""}, False, id="env_empty"),
+        param({"HYDRA_DEPRECATION_WARNINGS_AS_ERRORS": "1"}, True, id="env_set"),
+    ],
+)
+def test_deprecation_warning(env: Dict[str, str], expected_error: bool) -> None:
+    msg = "Feature FooBar is deprecated"
+    with patch("os.environ", env):
+        if expected_error:
+            with raises(DeprecationWarning, match=re.escape(msg)):
+                deprecation_warning(msg)
+        else:
+            with warns(UserWarning, match=re.escape(msg)):
+                deprecation_warning(msg)
```

Comment:
1. We use monkeypatch in other places. e.g. `monkeypatch.setenv("TEST_ENV", "test_env")`.
2. Can you add a test_app that calls `deprecation_warning()` and test that is fails with a proper error message?

Reply:
1. Got it. Updated to use `monkeypatch` in https://github.com/facebookresearch/hydra/pull/1703/commits/9931989b9725c8e40d3eb069e074b1a3efbbf3de.
2. In https://github.com/facebookresearch/hydra/pull/1703/commits/ea54062c149030e10321e7b94f2bcc6cc17b5ab9 I added a test app.